tenpy/simulations/time_evolution.py

```diff
@@ -74,3 +82,411 @@ def final_measurements(self):
+    """
+
+    def __init__(self, options, *, ground_state_data=None, ground_state_filename=None, **kwargs):
+        super().__init__(options, **kwargs)
+
+        resume_data = kwargs.get("resume_data", None)
+        if resume_data is not None:
+            if 'psi_ground_state' in resume_data:
+                self.psi_ground_state = resume_data['psi_ground_state']
+            else:
+                self.logger.warning("psi_ground_state not in resume data")
+            if 'gs_energy' in resume_data:
+                self.gs_energy = resume_data['gs_energy']
+            else:
+                self.logger.warning("ground-state energy not in resume data")
+
+        if not self.loaded_from_checkpoint:
+            if ground_state_filename is None:
+                ground_state_filename = self.options.get('ground_state_filename', None)
+            if ground_state_data is None and ground_state_filename is not None:
+                self.logger.info(
+                    f"loading data from 'ground_state_filename'='{ground_state_filename}'")
+                ground_state_data = hdf5_io.load(ground_state_filename)
+            elif ground_state_data is not None and ground_state_filename is not None:
+                self.logger.warning(
+                    "Supplied a 'ground_state_filename' and ground_state_data as kwarg. "
+                    "Ignoring 'ground_state_filename'.")
+
+            if ground_state_data is not None:
+                self.logger.info("Initializing from ground state data")
+                self._init_from_gs_data(ground_state_data)
+
+        # will be read out in init_state
+        self.gs_energy = self.options.get('gs_energy', None)
+        self.operator_t = self.options['operator_t']
+        # generate info for operator before time evolution as subconfig
+        self.operator_t0_config = self.options.subconfig('operator_t0')
+        self.operator_t0_name = self._get_operator_t0_name()
+        self.operator_t0 = None  # read out config later, since defaults depend on model parameters
+        self.addJW = self.options.get('addJW', False)
+
+    def resume_run(self):
+        if not hasattr(self, 'psi_ground_state'):
+            # didn't get psi_ground_state in resume_data, but might still have it in the results
+            if 'psi_ground_state' not in self.results:
+                raise ValueError("psi_ground_state not saved in checkpoint results: can't resume!")
+        super().resume_run()
+
+    def get_resume_data(self):
+        resume_data = super().get_resume_data()
+        resume_data['psi_ground_state'] = self.psi_ground_state
+        resume_data['gs_energy'] = self.gs_energy
+        return resume_data
+
+    def _connect_measurements(self):
+        """Connect :func:`m_correlation_function` to measurements."""
+        self._connect_measurements_fct('simulation_method', 'm_correlation_function', priority=1)
```

Comment:
why do we do it like this, and not add 
`('simulation_method', 'm_correlation_function')` to the class _default_measurements instead?
See the update of the defaul_measurements in RealTimeEvolution as an example

Reply:
In case `Simulation.use_default_measurements` is set to `False` the correlation function would still 
be measured. Otherwise when disabling the default measurements the class  `TimeDependentCorrelation` would 
still work but w/o any useful output. I'll add this to the docstring.